include/seastar/core/semaphore.hh

```diff
@@ -323,18 +357,41 @@ basic_semaphore<ExceptionFactory, Clock>::broken(std::exception_ptr xp) {
     }
 }
 
+#ifdef SEASTAR_DEADLOCK_DETECTION
```

Comment:
These lines look outright invalid for me. Why not leave the defaults as they are?